### PR TITLE
Filter: Improve method of loading required filter pkgs

### PR DESCRIPTION
**Pull request by pat-s**
As @mb706 	pointed out, this should be handled (for filters) in 

https://github.com/mlr-org/mlr/blob/f2817efb5d033adf2d090cd435852c974f1722d1/R/generateFilterValues.R#L41-L42

However,issue https://github.com/mlr-org/mlr/issues/2484 applies.

This PR solves the issue by

- setting the `pkg` slot of a filter that requires no pkg to be loaded to `NULL`.
- changing the if statement from above so that it actually works

### CODE REVIEW

**Comment by mb706** [comment]
Doesn't the `pkg  = "FSelectorRcpp",` argument in the `makeFilter` call already cause the package to be loaded when the filter is being used?

**Comment by berndbischl** [comment]
I don't like this custom change. We should load packages always in the same way

**Comment by pat-s** [comment]
@mb706 I need to check this.

@berndbischl we already use this way of loading the NS in the pkg. On the phone atm so can't refer to an example.

Just discovered today that all these msg are really annoying during usage.

**Comment by berndbischl** [comment]
Ok that's new to me? Please link to code

**Comment by berndbischl** [comment]
Actually please explain the problem first. That should always be done for a PR...

**Comment by mb706** [comment]
See #2484, apparently `requirePackages` *should* be called, but *isn't*.

**Comment by pat-s** [comment]
Take a look at https://github.com/mlr-org/mlr/issues/2484#issuecomment-438668903.

**Comment by pat-s** [comment]
Not yet finished/working. Will continue after the meeting.

**Comment by pat-s** [comment]
>  Also, since a user can define custom filters relying on other packages, the set of possible packages should not be enforced.

That's a good point. I changed it back to `assert_character(null.ok = TRUE)`.

**Comment by berndbischl** [comment]
I don't get the change. See Martins last comment. Why should packages be Null? And why should Nas be allowed? Packages should be a char vec, without Nas. If empty no packages are required. If not empty all packs are required

**Comment by pat-s** [comment]
> Why should packages be Null? And why should Nas be allowed?

Packages are still a chr vec. The only thing that changed is that filters that require NO pkg to be loaded (i.e. these pkgs are in base-R and loaded by default) now do not have a slot `pkg = character(0)` but `pkg = NULL` instead.

That's why I changed the assert function to accept `NULL`. `NULL` is imo more clean than using `character(0)` to me.
If we do only want to accept chrs, we should explicitly load the respective pkgs even if they are already loaded, e.g. for filter `variance` the `stats` pkg. 


The param decription of `pkg` in `makeFilter()` is: "Source package where the filter is implemented."
To me it does not make sense why this should evaluate to `character(0)` in any case?

File: R/Filter.R [hunk 1]
```diff
@@ -525,26 +525,26 @@ univariate = makeFilter(
...
-    fns = getTaskFeatureNames(task)
-    res = double(length(fns))
-    for (i in seq_along(fns)) {
-      subtask = subsetTask(task, features = fns[i])
-      res[i] = resample(learner = perf.learner, task = subtask, resampling = perf.resampling, measures = perf.measure, keep.pred = FALSE, show.info = FALSE)$aggr
-    }
-    if (perf.measure[[1L]]$minimize)
-      res = -1.0 * res
-    setNames(res, fns)
+      if (is.null(perf.measure))
```
Comment (mb706) [inline R/Filter.R]: Changing indentation here is probably going to be misleading.

**Comment by berndbischl** [comment]
> Packages are still a chr vec. The only thing that changed is that filters that require NO pkg to be loaded (i.e. these pkgs are in base-R and loaded by default) now do not have a slot `pkg = character(0)` but `pkg = NULL` instead.

i still dont get why we do that? as this is not consistent with what we do in learners? 
character(0) should be used if no packages are supposed to be loaded?

**Comment by pat-s** [comment]
> i still dont get why we do that?

I found it more logic to set a not-existend value to `NULL` rather than `character(0)`. But if we introduce inconsistency with that, fair enough :)

@berndbischl 	tidied up everything, ready-to-go now?

**Comment by pat-s** [comment]
ping @berndbischl

**Pull request merged**